lightning/src/ln/offers_tests.rs

```diff
@@ -2276,11 +2276,19 @@ fn fails_paying_invoice_with_unknown_required_features() {
 	let secp_ctx = Secp256k1::new();
 
 	let created_at = alice.node.duration_since_epoch();
-	let invoice = invoice_request
-		.verify_using_recipient_data(nonce, &expanded_key, &secp_ctx).unwrap()
-		.respond_using_derived_keys_no_std(payment_paths, payment_hash, created_at).unwrap()
-		.features_unchecked(Bolt12InvoiceFeatures::unknown())
-		.build_and_sign(&secp_ctx).unwrap();
+	let verified_invoice_request = invoice_request
```

Comment:
For review, I'd prefer all commits up to this one to be squashed. The pattern of 'rename - add new type - replace usages' makes it harder for me to see what the actual change is. This is what I am looking for in a diff:

<img width="1327" height="544" alt="image" src="https://github.com/user-attachments/assets/549b54ed-38eb-40d8-ae81-f9fa5e70f6ee" />